variants/board_template/variant.h

```diff
@@ -162,6 +162,21 @@ extern "C" {
 // See AN4879 https://www.st.com/content/st_com/en/search.html#q=AN4879-t=resources-page=1
 //#define USBD_VBUS_DETECTION_ENABLE
 
+// If the board has external USB pullup (on DP/DM depending on speed)
+// that can be controlled using a GPIO pin, define these:
+//  - If the the pullup is disabled (USB detached) by default, define
+//    USBD_ATTACH_PINNAME to the pin that, when written to
+//    USBD_ATTACH_LEVEL, attaches the pullup.
+//  - If the the pullup is enabled (attached) by default, define
+//    USBD_DETACH_PINNAME to the pin that, when written to
+//    USBD_DETACH_LEVEL, detaches the pullup.
+// Note that these constants need PinName values (e.g. PB_5) not pin
+// numbers (e.g. PB5).
+//#define USBD_ATTACH_PINNAME x
```

Comment:
To reinforce your note, x could be advantageously replaced by PY_x format used in wiki for example
```suggestion
//#define USBD_ATTACH_PINNAME PY_x
```